File: Day1.fs
module AdventOfCode.Day1

let numbers = Input.forDay 1 |> Array.map int

let part1 =
    [ for n1 in numbers do
        for n2 in numbers do
            if n1 + n2 = 2020 then yield n1 * n2 ]
    |> List.head

let part2 =
    [ for n1 in numbers do
        for n2 in numbers do
            for n3 in numbers do
                if n1 + n2 + n3 = 2020 then yield n1 * n2 * n3 ]
    |> List.head

let solution = part1, part2
    

File: 2021/src/day1.nim
import helpers

proc solveDay1*: IntSolution =
  let depths = readInputIntSeq("input/day1.txt")

  for i in 0 .. depths.high:
    if i < depths.high and depths[i + 1] > depths[i]:
      inc result.part1
    if i < depths.high - 2 and depths[i + 3] > depths[i]:
      inc result.part2

when isMainModule:
  echo solveDay1()
